metabase/metabase
# Browsing | Fix entering/leaving admin breaks app layout

Entering and leaving the admin app was breaking the app layout. It was caused by `fitViewport` HOC removing `App` component's top-level `flex` CSS class ([see here](https://github.com/metabase/metabase/blob/6d160a093d4ed6e5a9e12037fa9e60103bfbf2e7/frontend/src/metabase/hoc/FitViewPort.jsx#L56)). Fixed by replacing `App` component's visual components with `@emotion/styled` components (something we should have done anyway). In that way, there is no `flex` class to be removed by the `fitViewport` HOC, and the styling is managed in React components for now

### To Verify

1. Sign in as admin
2. Open the core Metabase app (e.g. just the homepage `/`)
3. Click the "gear" icon at the top right, click "Admin settings"
4. Click "Exit admin" at the top right
5. Make sure you land on the homepage and the UI isn't broken
6. Scroll the homepage down to the list of connected databases and click "Add database"
7. Ensure the admin UI is not broken

### Demo

**Before**

https://user-images.githubusercontent.com/17258145/158886128-9058e7e8-770d-4567-bd39-cafc2277c826.mp4

**Demo**

https://user-images.githubusercontent.com/17258145/158886148-e9d845e5-b310-4738-923a-a04a4ea623be.mp4
